rust/hook-janitor/src/webhooks.rs

```diff
@@ -139,6 +134,124 @@ impl From<FailedRow> for AppMetric {
     }
 }
 
+#[derive(sqlx::FromRow, Debug)]
+struct HoghookCompletedRow {
+    // App Metrics truncates/aggregates rows on the hour, so we take advantage of that to GROUP BY
+    // and aggregate to select fewer rows.
+    hour: DateTime<Utc>,
+    // A note about the `try_from`s: Postgres returns all of those types as `bigint` (i64), but
+    // we know their true sizes, and so we can convert them to the correct types here. If this
+    // ever fails then something has gone wrong.
+    #[sqlx(try_from = "i64")]
+    team_id: u32,
+    app_source_id: String,
+    #[sqlx(try_from = "i64")]
+    count: u32,
+}
+
+impl From<HoghookCompletedRow> for AppMetric2 {
+    fn from(row: HoghookCompletedRow) -> Self {
+        AppMetric2 {
+            team_id: row.team_id,
+            timestamp: row.hour,
+            app_source: app_metrics2::Source::Hoghooks,
+            app_source_id: row.app_source_id,
+            instance_id: None,
+            metric_kind: app_metrics2::Kind::Success,
+            metric_name: "".to_owned(),
+            count: row.count,
+        }
+    }
+}
+
+#[derive(sqlx::FromRow, Debug)]
+struct HoghookFailedRow {
+    // App Metrics truncates/aggregates rows on the hour, so we take advantage of that to GROUP BY
+    // and aggregate to select fewer rows.
+    hour: DateTime<Utc>,
+    // A note about the `try_from`s: Postgres returns all of those types as `bigint` (i64), but
+    // we know their true sizes, and so we can convert them to the correct types here. If this
+    // ever fails then something has gone wrong.
+    #[sqlx(try_from = "i64")]
+    team_id: u32,
+    app_source_id: String,
+    #[sqlx(try_from = "i64")]
+    count: u32,
+}
+
+impl From<HoghookFailedRow> for AppMetric2 {
+    fn from(row: HoghookFailedRow) -> Self {
+        AppMetric2 {
+            team_id: row.team_id,
+            timestamp: row.hour,
+            app_source: app_metrics2::Source::Hoghooks,
+            app_source_id: row.app_source_id,
+            instance_id: None,
+            metric_kind: app_metrics2::Kind::Failure,
+            metric_name: "".to_owned(),
```

Comment:
I think these should have a value as well? 

From what we discussed before I saw "kind" as generic and ideally common across multiple services whereas `name` is more specific to what metric is being reported. Having this empty feels confusing...

Reply:
I struggled here, would you just repeat what's in kind (which is already `success` or `failure`)?

The `app_source` is already `hoghooks` and I'm already saying `success`, and `hoghooks` functions only do one thing -- I had no clue what else to say about this metric. I guess I could put in `fetch` (which feels redundant with `hoghooks`)?

Reply:
Oh, the only idea I had yesterday is to put like `http:${statusCode}` or something...

Reply:
Yeah thats already way more useful I would argue